### ISSUE TITLE
Distributed ACID transactions 

### ISSUE THREAD

**Issue by jpiquot** [dapr/dotnet-sdk#282]
Hi,
In some scenarios, we need multiple actors to join a transaction regardless of where their state is stored. 

The only information I found on transactions are in this document :
[API-005-state-store-behavior](https://github.com/dapr/dapr/blob/master/docs/decision_records/api/API-005-state-store-behavior.md)

It does not say much on how this will be implemented in Dapr .NET.

**Comment by amanbha** [comment]
@jpiquot The transaction is to update state per method call of a particular actor.
The transactions cannot span multiple actor methods.

**Comment by jpiquot** [comment]
@amanbha Yes, it is an anti pattern, if we use transactions everywhere. But in some cases we may need ACID transactions. Some actor frameworks support transactions like the Microsoft Orleans project :  
- [Microsoft Orleans](https://github.com/dotnet/orleans). 
- [Orleans Transactions](https://dotnet.github.io/orleans/Documentation/grains/transactions.html).

> Orleans Distributed ACID Transactions :
> In addition to the simple persistence model described above, grains can have transactional state. Multiple grains can participate in ACID transactions together regardless of where their state is ultimately stored. Transactions in Orleans are distributed and decentralized (there is no central transaction manager or transaction coordinator) and have serializable isolation. For more information on transactions in Orleans, see the documentation and the Microsoft Research technical report.
>

**Comment by amanbha** [comment]
@jpiquot It can be treated as a proposal, If we decide to support it, It needs to be done in Dapr runtime and not at individual SDK level.
 I am transferring the issue to dapr repo. Please feel free to add to it on how to expose it at Dapr level,
